compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DocumentationAnalyzer.java

```diff
@@ -210,28 +210,12 @@ public void visit(BLangTypeDefinition typeDefinition, AnalyzerData data) {
         BLangType typeNode = typeDefinition.getTypeNode();
         if (typeDefinition.typeNode.getKind() == NodeKind.OBJECT_TYPE) {
             List<BLangSimpleVariable> fields = ((BLangObjectTypeNode) typeNode).fields;
-            validateParameters(typeDefinition, fields, null, DiagnosticWarningCode.UNDOCUMENTED_FIELD,
-                    DiagnosticWarningCode.NO_SUCH_DOCUMENTABLE_FIELD, DiagnosticWarningCode.FIELD_ALREADY_DOCUMENTED);
-            validateReturnParameter(typeDefinition, null, false);
-            validateReferences(typeDefinition, data);
-            for (BLangSimpleVariable field : fields) {
-                validateReferences(field, data);
-                validateDeprecationDocumentation(field.getMarkdownDocumentationAttachment(),
-                        Symbols.isFlagOn(field.symbol.flags, Flags.DEPRECATED),
-                        field.getPosition());
-            }
+            validateObjectOrRecord(typeDefinition, fields, data);
 
             ((BLangObjectTypeNode) typeDefinition.getTypeNode()).getFunctions().forEach(t -> visitNode(t, data));
         } else if (typeDefinition.typeNode.getKind() == NodeKind.RECORD_TYPE) {
             List<BLangSimpleVariable> fields = ((BLangRecordTypeNode) typeNode).fields;
-            validateParameters(typeDefinition, fields, null, DiagnosticWarningCode.UNDOCUMENTED_FIELD,
-                    DiagnosticWarningCode.NO_SUCH_DOCUMENTABLE_FIELD, DiagnosticWarningCode.FIELD_ALREADY_DOCUMENTED);
-            validateReturnParameter(typeDefinition, null, false);
-            validateReferences(typeDefinition, data);
-
-            for (SimpleVariableNode field : fields) {
-                validateReferences(field, data);
-            }
+            validateObjectOrRecord(typeDefinition, fields, data);
```

Comment:
Let's extract `typeDefinition.typeNode.getKind()` (say `kind`) out to a variable,  check `if (kind == NodeKind.OBJECT_TYPE || kind == NodeKind.RECORD_TYPE)`, and call the helper method by casting to `BLangStructureTypeNode` which both `BLangRecordTypeNode` and `BLangObjectTypeNode` extend.